src/status/gif.nim

```diff
@@ -0,0 +1,57 @@
+import httpclient
+import json
+import strformat
+import os
+
+# set via `nim c` param `-d:TENOR_API_KEY:[api_key]`; should be set in CI/release builds
+const TENOR_API_KEY {.strdefine.} = ""
+let TENOR_API_KEY_ENV = $getEnv("TENOR_API_KEY")
+
+let TENOR_API_KEY_RESOLVED =
+  if TENOR_API_KEY_ENV != "":
+    TENOR_API_KEY_ENV
+  else:
+    TENOR_API_KEY
+
+const baseUrl = "https://g.tenor.com/v1/"
+let defaultParams = fmt("&media_filter=basic&limit=10&key={TENOR_API_KEY_RESOLVED}")
+
+type
+  GifItem* = object
+    id*: int
+    title*: string
+    url*: string
+
+proc toGifItem(jsonMsg: JsonNode): GifItem =
+  return GifItem(
+    id: jsonMsg{"id"}.getInt,
+    title: jsonMsg{"title"}.getStr,
+    url: jsonMsg{"media"}[0]["gif"]["url"].getStr
+  )
+
+proc `$`*(self: GifItem): string =
+  return fmt"GifItem(id:{self.id}, title:{self.title}, url:{self.url})"
+
+type
+  GifClient* = ref object
+    client: HttpClient
+
+proc newGifClient*(): GifClient =
+  result = GifClient()
+  result.client = newHttpClient()
+
+proc tenorQuery(self: GifClient, path: string): seq[GifItem] = 
+  let content = self.client.getContent(fmt("{baseUrl}{path}{defaultParams}"))
```

Comment:
I think this might need some error handling. What happens if the query fails?

Reply:
If it fails..... :bomb: Would it be ok for you to retry n times, let's say 3 and if after that still nothing then return an empty seq?

Reply:
I think for now, just return the empty seq when it fails. We have a couple of other API calls that don't have retries, so maybe we'd want to add retries to all of them or just keep to none?
I don't know. 🤷